### PR TITLE
Ard chapter08

### PR DESCRIPTION
**Pull request by ardavey**

### CODE REVIEW

File: chapter08/ard/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use IO::Handle;
+use IO::Scalar;
+use IO::Tee;
+
+# nasty globals, but will do for now
+my $output_handle; # this will ultimately point to our destination(s)
+my $scalar;
+
+my $t = localtime;
+my $output_string = sprintf( "Today is %s %u %s %u", $t->fullday, $t->day_of_month, $t->fullmonth, $t->year );
```
Comment (utopia-2) [inline chapter08/ard/ex1.pl:17]: :+1: Nice! I'll have to give `Time::Piece` a try at some point.

File: chapter08/ard/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+chomp( my $choice = lc( <STDIN> ) );
+#$choice = lc( $choice );
+
+my %dispatch = (
+  f => \&do_file,
+  s => \&do_scalar,
+  b => \&do_both,
+);
+
+( $dispatch{$choice} || \&do_invalid )->();
```
Comment (utopia-2) [inline chapter08/ard/ex1.pl:30]: Why not return the handle from these functions rather than assigning to the global?

File: chapter08/ard/ex1.pl [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+print "Output to file, scalar or both? [fsb] ";
+chomp( my $choice = lc( <STDIN> ) );
+#$choice = lc( $choice );
+
+my %dispatch = (
+  f => \&do_file,
+  s => \&do_scalar,
+  b => \&do_both,
+);
```
Comment (utopia-2) [inline chapter08/ard/ex1.pl:28]: \o/ dispatch table. cgy would be pleased.

File: chapter08/ard/ex3.pl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+foreach my $arg ( @ARGV ) {
+  if ( -d $arg ) {
+    push @dirs, $arg;
+  }
+  else {
+    say "'$arg' is not a valid directory";
+  }
+}
+
+my @dir_handles = map { my $dh = IO::Dir->new( $_ ); $dh } @dirs;
```
Comment (utopia-2) [inline chapter08/ard/ex3.pl:20]: Could simplify this to `map { IO::Dir->new( $_ ) } @dirs`, or even `map IO::Dir->new( $_ ), @dirs` (though the latter can be confusing).

File: chapter08/ard/ex3.pl [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+for ( my $i = 0; $i <= $#dirs; $i++ ) {
```
Comment (utopia-2) [inline chapter08/ard/ex3.pl:22]: Old-school for loop? You don't appear to use the index anywhere?

File: chapter08/ard/ex3.pl (hunk 4 above)
Comment (JPeace2) [inline chapter08/ard/ex3.pl:20]: I think the exercise required using opendir to create the directory handle.

File: chapter08/ard/ex3.pl (hunk 5 above)
Comment (ardavey) [inline chapter08/ard/ex3.pl:22]: Only in every line inside the loop...

Comment (utopia-2) [inline chapter08/ard/ex3.pl:22]: Ah-hah, two arrays, my mistake!

Comment (ardavey) [inline chapter08/ard/ex3.pl:22]: Yeah, this was my slightly less elegant approach to allow inline printing of the directory names without using a hash with weird keys.

File: chapter08/ard/ex1.pl (hunk 1 above)
Comment (ardavey) [inline chapter08/ard/ex1.pl:17]: It's _incredibly_ useful!

File: chapter08/ard/ex1.pl (hunk 3 above)
Comment (ardavey) [inline chapter08/ard/ex1.pl:28]: That was my primary goal.

File: chapter08/ard/ex1.pl (hunk 2 above)
Comment (ardavey) [inline chapter08/ard/ex1.pl:30]: Yeah, that's a good point.  I'm going to chalk that up as a consequence of completely re-architecting the script after already implementing a different solution.

**Pull request merged**